app/controllers/asset/noneeabusiness/remove/RemoveAssetYesNoController.scala

```diff
@@ -88,15 +92,23 @@ class RemoveAssetYesNoController @Inject()(
                 }
             }
           } else {
-            Future.successful(redirectToAddAssetsPage)
+            if (request.userAnswers.isMigratingToTaxable) {
+              Future.successful(Redirect(controllers.asset.nonTaxableToTaxable.routes.AddAssetsController.onPageLoad()))
+            } else {
+              Future.successful(Redirect(controllers.asset.noneeabusiness.routes.AddNonEeaBusinessAssetController.onPageLoad()))
+            }
           }
         }
       )
   }
 
-  private def removeAsset(identifier: String, index: Int)(implicit hc: HeaderCarrier): Future[Result] = {
+  private def removeAsset(identifier: String, index: Int)(implicit request: DataRequest[AnyContent]): Future[Result] = {
     trustService.removeAsset(identifier, RemoveAsset(AssetNameType.NonEeaBusinessAssetNameType, index)).map(_ =>
-      redirectToAddAssetsPage
+      if (request.userAnswers.isMigratingToTaxable) {
+        Redirect(controllers.asset.nonTaxableToTaxable.routes.AddAssetsController.onPageLoad())
+      } else {
+        Redirect(controllers.asset.noneeabusiness.routes.AddNonEeaBusinessAssetController.onPageLoad())
+      }
```

Comment:
Could this be extracted to its own def? Same logic used 3 times in this controller